### ISSUE TITLE
wrong order in tf_model_loss

### ISSUE THREAD

**Issue by hengyuan-hu** [openai/cleverhans#25]
In utils_tf.py, line 26 and 29, it should be categorical_crossentropy(model, y) instead of categorical_crossentropy(y, model). This won't affect training in a noticeable way for mnist but will make the training fail to work when I tried a deeper model on cifar-10.

**Comment by goodfeli** [comment]
Thanks!
Also, it looks like keras defaults to using the numerically inaccurate calculation of the cross-entropy using the probabilities rather than the logits. I will fix that too.
